src/mmj/transforms/EquivalenceInfo.java

```diff
@@ -88,11 +88,11 @@ public void fillDeductRules(final List<Assrt> assrtList,
             final Map<Stmt, Assrt> dedCom = eqDeductCom.get(op);
             final Map<Stmt, Assrt> dedTrans = eqDeductTrans.get(op);
             for (final Stmt eqOp : eqCommutatives.keySet()) {
-                if (!dedCom.containsKey(eqOp))
+                if (dedCom == null || !dedCom.containsKey(eqOp))
```

Comment:
I think these checks should be hoisted out of the loop, after the creation of `dedCom` and `dedTrans`.